integration_test.go

```diff
@@ -1012,6 +1011,71 @@ func TestFsnotifyClose(t *testing.T) {
 	}
 }
 
+func TestSymlinkNotFollowed(t *testing.T) {
+	testDir := tempMkdir(t)
+	file1 := filepath.Join(testDir, "file1")
+	file2 := filepath.Join(testDir, "file2")
+	link := filepath.Join(testDir, "link")
+
+	f1, err := os.Create(file1)
+	if err != nil {
+		t.Fatalf("Failed to create file1: %s", err)
+	}
+	defer f1.Close()
+	if _, err := os.Create(file2); err != nil {
+		t.Fatalf("Failed to create file2: %s", err)
+	}
+
+	// symlink works for Windows too
+	if err := os.Symlink(file1, link); err != nil {
+		t.Fatalf("Failed to create sylink: %s", err)
+	}
+
+	w, err := NewWatcher()
+	if err != nil {
+		t.Fatalf("Failed to create watcher")
+	}
+
+	err = w.Add(link)
+	if err != nil {
+		t.Fatalf("Failed to add link: %s", err)
+	}
+
+	// change file 1 - no event
+	f1.Write([]byte("Hello"))
+	f1.Sync()
+	// XXX(obristow): doing a create here shows a CHMOD event on mac - is that an issue?
+
+	select {
+	case event := <-w.Events:
+		t.Fatalf("Event from watcher: %v", event)
+	case err := <-w.Errors:
+		t.Fatalf("Error from watcher: %v", err)
+	case <-time.After(50 * time.Millisecond):
+	}
+
+	// ~atomic link change - event
+	tmpLink := filepath.Join(testDir, "tml-link")
+	if err := os.Symlink(file2, tmpLink); err != nil {
+		t.Fatalf("Failed to create sylink: %s", err)
+	}
+
+	if err := os.Rename(tmpLink, link); err != nil {
+		t.Fatalf("Failed to replace sylink: %s", err)
+	}
+
+	select {
+	case _ = <-w.Events:
+	case err := <-w.Errors:
+		t.Fatalf("Error from watcher: %v", err)
+	case <-time.After(50 * time.Millisecond):
+		t.Fatalf("Took too long to wait for event")
+	}
+
+}
+
+/* XXX(obristow): these would return if implemeting the high level interfaced mentioned in https://github.com/fsnotify/fsnotify/issues/199#issuecomment-480277398
```

Comment:
Should we remove this test, or keep it and use EvalSymlinks?

Reply:
I pushed an amended version with these tests stripped, as I'm thinking we can dig them out of history if we want, and I don't think they'd be very meaningful if using `EvalSymlinks`.

Reply:
I've just reinstated these tests now that `watcher.Add(name)` is back to the symlink following behaviour